uint8 id
float64 x
float64 y
float64 vx
float64 vy